### PR TITLE
Replaced calls to old JSON reader/writer

### LINKED ISSUE #778: [FEATURE] Remove uses of old JsonWriter
`SanyParserPass` still uses the old `JsonWriter` to output `out-parse.json`. It should use the new implementation with the standardized JSON encoding (see #595).

### PR DESCRIPTION
**Pull request by Kukovec**
- [ ] ~Tests added for any new code~
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] ~Documentation added for any new functionality~
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

closes #778

### CODE REVIEW

**Review by konnov — COMMENTED** [review]
Looks good. Let's do a bit of refactoring to save our efforts in the future.

File: tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala [hunk 1]
```diff
@@ -22,8 +23,9 @@ import java.nio.file.Path
  * @author Igor Konnov
  */
 class SanyParserPassImpl @Inject() (
-    val options: PassOptions, val sourceStore: SourceStore, val annotationStore: AnnotationStore,
-    val writerFactory: TlaWriterFactory, @Named("AfterParser") val nextPass: Pass with TlaModuleMixin
+    val options: PassOptions, val sourceStore: SourceStore, val changeListener: ChangeListener,
```
[the remark this answers is not in the document]
Comment (Kukovec) [reply tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala:26]: `SourceLocator` needs it in its constructor.

File: tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala [hunk 2]
```diff
@@ -110,6 +115,17 @@ class SanyParserPassImpl @Inject() (
...
+    val writer = new PrintWriter(new FileWriter(file, false))
+    try {
+      val sourceLocator: SourceLocator = SourceLocator(sourceStore.makeSourceMap, changeListener)
+      val jsonText = new TlaToUJson(Some(sourceLocator)).makeRoot(Seq(module)).toString
+      writer.write(jsonText)
+    } finally {
+      writer.close()
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (Kukovec) [reply tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala:128]: Yeah, that should be easy enough

File: tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala [hunk 3]
```diff
@@ -44,7 +46,13 @@ class SanyParserPassImpl @Inject() (
     val filename = options.getOrError("parser", "filename").asInstanceOf[String]
     if (filename.endsWith(".json")) {
       try {
-        rootModule = Some(JsonReader.readModule(new File(filename)))
+        val moduleJson = UJsonRep(ujson.read(new File(filename)))
+        // TODO: Implement a TagReader
```
[the remark this answers is not in the document]
Comment (Kukovec) [reply tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala:50]: #780

[the remark this answers is not in the document]
Comment (konnov) [reply tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala:50]:
```suggestion
        // TODO: Implement a TagReader in issue #780
```

File: tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (konnov) [reply tla-import/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala:128]: Let's merge this PR. I have to touch the writers too.

**Review by konnov — APPROVED** [review]
Let's merge it as it is, to keep things moving.

**Pull request merged**